src/widgets/table.rs

```diff
@@ -286,14 +300,20 @@ impl<'a> Table<'a> {
         self
     }
 
+    /// Set which style of selection space allocation to use
+    pub fn highlight_set_selection_space(mut self, value: HighlightSpacing) -> Self {
```

Comment:
You pushed just as my last comment was written :)

In case you missed it:

> In the doc comment it would be good to note why this feature exists. Sort of reiterate the commit message but reword it so that it fits well with the tone of the docs - i.e. "this works like ..." rather than "this is changed because ..."

Putting my user hat on, this comment is difficult to understand. Perhaps something like:

```suggestion
    /// When to show the highlight spacing.
    ///
    /// A highlight symbol can be displayed to the left of the row. If there is no selected row, this setting allows
    /// space to be shown or omitted. If the space is omitted then the table columns will change size when...
    pub fn highlight_spacing(mut self, value: HighlightSpacing) -> Self {
```